integration/hibernate-4/src/main/java/com/blazebit/persistence/impl/hibernate/Hibernate4Integrator.java

```diff
@@ -36,7 +37,8 @@ public void integrate(Configuration configuration, SessionFactoryImplementor ses
 				// TODO: check that no collections are mapped
 			}
 		}
-		CommonHibernateUtils.setCollectionPersisterClass(configuration.getClassMappings(), CustomOneToManyPersister.class);
+
+		serviceRegistry.locateServiceBinding(PersisterClassResolver.class).setService(new CustomPersisterClassResolver());
```

Comment:
I've been doing some manual testing in my own project and for me it seems to work on Hibernate 5.2 as well! :smile: